log_analytics.tf

```diff
@@ -81,3 +81,65 @@ resource "azurerm_log_analytics_solution" "main" {
     publisher = "Microsoft"
   }
 }
+
+locals {
+  dcr_location = try(coalesce(try(local.log_analytics_workspace.location, null), try(data.azurerm_log_analytics_workspace.main[0].location, null)), null)
+}
+
+resource "azurerm_monitor_data_collection_rule" "dcr" {
+  count               = (var.log_analytics_workspace_enabled && var.oms_agent_enabled) ? 1 : 0
```

Comment:
Would `count               = (local.create_analytics_workspace && var.oms_agent_enabled) ? 1 : 0` be better?